packages/dx-react-chart/src/plugins/layout-manager.jsx

```diff
@@ -0,0 +1,245 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { Plugin, Getter } from '@devexpress/dx-react-core';
+import { TOP, BOTTOM, LEFT, RIGHT } from '@devexpress/dx-chart-core';
+import yoga, { Node } from '@devexpress/dx-flex-layout';
+
+const LayoutElement = () => null;
+
+const isEqual = (
+  { width: firstWidth, height: firstHeight },
+  { width: secondWidth, height: secondHeight },
+) => firstWidth === secondWidth && firstHeight === secondHeight;
+
+const createNode = ({
+  flexGrow, flexDirection, width, height, margin, alignItems, flexWrap, justifyContent,
+} = {}) => {
+  const node = Node.create();
+  node.setDisplay(yoga.DISPLAY_FLEX);
+  if (flexGrow) node.setFlexGrow(flexGrow);
+  if (flexDirection) node.setFlexDirection(flexDirection);
+  if (width) node.setWidth(width);
+  if (height) node.setHeight(height);
+  if (margin) node.setMargin(yoga.EDGE_ALL, margin);
+  if (alignItems) node.setAlignItems(alignItems);
+  if (flexWrap) node.setFlexWrap(flexWrap);
+  if (justifyContent) node.setJustifyContent(justifyContent);
+  return node;
+};
+
+const getPosition = (node, positions, methodName) => {
+  const parent = node.getParent();
+  if (parent) {
+    return getPosition(parent, (positions + parent[methodName]()), methodName);
+  }
+  return positions;
+};
+
+const getXPosition = (node, positions) => getPosition(node, positions, 'getComputedLeft');
+const getYPosition = (node, positions) => getPosition(node, positions, 'getComputedTop');
+
+const getAbsoluteNodePosition = node => ({
+  x: getXPosition(node, node.getComputedLeft()),
+  y: getYPosition(node, node.getComputedTop()),
+  width: node.getComputedWidth(),
+  height: node.getComputedHeight(),
+});
+
+const calculatePositions = (rootNode, width, height, nodes) => {
+  const positions = {};
+  rootNode.calculateLayout(width, height, yoga.DIRECTION_LTR);
+  nodes.forEach(({ node }, name) => {
```

Comment:
We prefer functional approach. You can use `Array.reduce` to fill the `positions` object.